network/p2p/unicast/manager.go

```diff
@@ -138,49 +244,108 @@ func (m *Manager) rawStreamWithProtocol(ctx context.Context,
 		// cancel the dial back off (if any), since we want to connect immediately
 		dialAddr = m.streamFactory.DialAddress(peerID)
 		m.streamFactory.ClearBackoff(peerID)
-
-		// if this is a retry attempt, wait for some time before retrying
-		if retries > 0 {
-			// choose a random interval between 0 to 5
-			// (to ensure that this node and the target node don't attempt to reconnect at the same time)
-			r := rand.Intn(MaxConnectAttemptSleepDuration)
-			time.Sleep(time.Duration(r) * time.Millisecond)
-		}
-
 		err := m.streamFactory.Connect(ctx, peer.AddrInfo{ID: peerID})
 		if err != nil {
-
 			// if the connection was rejected due to invalid node id, skip the re-attempt
 			if strings.Contains(err.Error(), "failed to negotiate security protocol") {
-				return s, dialAddr, fmt.Errorf("invalid node id: %w", err)
+				return fmt.Errorf("failed to dial remote peer: %w", err)
 			}
 
 			// if the connection was rejected due to allowlisting, skip the re-attempt
 			if errors.Is(err, swarm.ErrGaterDisallowedConnection) {
```

Comment:
would be great to document this sentinel as a potential return of `StreamFactory.Connect`

Reply:
https://github.com/onflow/flow-go/pull/3902/commits/836d0b6898b147859309c965ec60b308e2720515